deno: chokes on `derivatives/bagel.csv`

AFAIK we do not mandate naming under `derivatives/`. But I am getting

```shell
❯ bids-validator-deno $PWD
	[ERROR] Files with such naming scheme are not part of BIDS specification. This error is most commonly caused by typos in file names that make them not BIDS compatible. Please consult the specification and make sure your files are named correctly. If this is not a file naming issue (for example when including files not yet covered by the BIDS specification) you should include a ".bidsignore" file in your dataset (see https://github.com/bids-standard/bids-validator#bidsignore for details). Please note that derived (processed) data should be placed in /derivatives folder and source data (such as DICOMS or behavioural logs in proprietary formats) should be placed in the /sourcedata folder. (NOT_INCLUDED)

		./derivatives/bagel.csv

		1 more files with the same issue

	Please visit https://neurostars.org/search?q=NOT_INCLUDED for existing conversations about this issue.
```

--- a/bids-validator/src/validators/bids.ts
+++ b/bids-validator/src/validators/bids.ts
@@ -62,21 +62,27 @@ export async function validate(
     issues.addNonSchemaIssue('MISSING_DATASET_DESCRIPTION', [] as IssueFile[])
   }
 
-  let derivatives: FileTree[] = []
+  const bidsDerivatives: FileTree[] = []
+  const nonstdDerivatives: FileTree[] = []
   fileTree.directories = fileTree.directories.filter((dir) => {
-    if (dir.name === 'derivatives') {
-      dir.directories.map((deriv) => {
-        if (
-          deriv.files.some(
-            (file: BIDSFile) => file.name === 'dataset_description.json',
-          )
-        ) {
-          derivatives.push(deriv)
-        }
-      })
+    if (dir.name !== 'derivatives') {
       return true
     }
-    return true
+    for (const deriv of dir.directories) {
+      if (
+        deriv.files.some(
+          (file: BIDSFile) => file.name === 'dataset_description.json',
+        )
+      ) {
+        // New root for the derivative dataset
+        deriv.parent = undefined
+        bidsDerivatives.push(deriv)
+      } else {
+        nonstdDerivatives.push(deriv)
+      }
+    }
+    // Remove derivatives from the main fileTree
+    return false
   })
 
   for await (const context of walkFileTree(fileTree, issues, dsContext)) {
@@ -103,7 +109,7 @@ export async function validate(
 
   let derivativesSummary: Record<string, ValidationResult> = {}
   await Promise.allSettled(
-    derivatives.map(async (deriv) => {
+    bidsDerivatives.map(async (deriv) => {
       derivativesSummary[deriv.name] = await validate(deriv, options)
       return derivativesSummary[deriv.name]
     }),

